frontend/packages/noobaa-storage-plugin/src/components/data-consumption-card/data-consumption-card-utils.ts

```diff
@@ -0,0 +1,149 @@
+import { getMetric, getValue, PrometheusMetricResult } from '../../utils';
+import {
+  PROVIDERS,
+  ACCOUNTS,
+  BY_IOPS,
+  BY_LOGICAL_USAGE,
+  BY_PHYSICAL_VS_LOGICAL_USAGE,
+  BY_EGRESS,
+} from '../../constants';
+
+/* Chart Data Handlers */
+
+const iopsChartData: GetBarChartData = (data, metricType, metric1, metric2) => {
+  const arr1 = [];
+  const arr2 = [];
+  _.forEach(data, (iops) => {
+    const y1 = getMetric(iops, metric1);
+    const y2 = getMetric(iops, metric2);
+    const reads = {
+      name: 'Total Reads',
+      x: getMetric(iops, metricType),
+      y: Number(humanizeDecimalBytesWithoutB(y1).value),
+    };
+    const writes = {
+      name: 'Total Writes',
+      x: getMetric(iops, metricType),
+      y: Number(humanizeDecimalBytesWithoutB(y2).value),
+    };
+    arr1.push(reads);
+    arr2.push(writes);
+  });
+  return arr1.length > 0 && arr2.length > 0 ? [arr1, arr2] : [];
+};
+
+const accountsLogicalUsageChartData: GetBarChartData = (data, metricType) => [
+  data.map((logicalUsage) => ({
+    name: 'Logical Usage',
+    x: getMetric(logicalUsage, metricType),
+    y: Number(humanizeBinaryBytesWithoutB(getValue(logicalUsage)).value),
+  })),
+];
+const providersPhysicalVsLogicalChartData: GetBarChartData = (data) => {
+  const arr1 = [];
+  const arr2 = [];
+  _.forEach(data, (physicalVsLogicalData) => {
+    const y1 = Number(getMetric(physicalVsLogicalData, 'physical_size'));
+    const y2 = Number(getMetric(physicalVsLogicalData, 'logical_size'));
+    const reads = {
+      name: 'Total Physical',
+      x: getMetric(physicalVsLogicalData, 'type'),
+      y: Number(humanizeDecimalBytesWithoutB(y1).value),
+    };
+    const writes = {
+      name: 'Total Logical',
+      x: getMetric(physicalVsLogicalData, 'type'),
+      y: Number(humanizeDecimalBytesWithoutB(y2).value),
+    };
+    arr1.push(reads);
+    arr2.push(writes);
+  });
+  return arr1.length > 0 && arr2.length > 0 ? [arr1, arr2] : [];
+};
+
+const providersEgressChartData: GetBarChartData = (data) => [
+  data.map((egress) => ({
+    name: getMetric(egress, 'type'),
+    x: getMetric(egress, 'type'),
+    y: Number(humanizeBinaryBytesWithoutB(getValue(egress)).value),
+  })),
+];
+
+/* Legends Data Handlers */
+
+const iopsChartLegendData = (data: [BarChartData[], BarChartData[]]) => {
+  const totalReads = data[0].reduce((total: number, iops: BarChartData) => total + iops.y, 0);
+  const totalWrites = data[1].reduce((total: number, iops: BarChartData) => total + iops.y, 0);
+  return [{ name: `Total Reads ${totalReads}` }, { name: `Total Writes ${totalWrites}` }];
+};
+
+const providersPhysicalVsLogicalChartLegendData = (data: [BarChartData[], BarChartData[]]) => {
+  const valueL = humanizeBinaryBytesWithoutB(
+    data[0].reduce((total: number, logical: BarChartData) => total + logical.y, 0),
+  );
+  const valueP = humanizeBinaryBytesWithoutB(
+    data[1].reduce((total: number, physical: BarChartData) => total + physical.y, 0),
+  );
```

Comment:
Nit: Here also you can destruct the valueL and valueP as you did for others `humanizeBinaryBytesWithoutB` return value.

Reply:
nope I cant because I am using it multiple times, there will be same variables